source/framework/CMakeLists.txt

```diff
@@ -17,51 +17,75 @@ configure_file(
   @ONLY
 )
 
+# MarlinCore library
 aux_source_directory( src MarlinCore_sources )
 aux_source_directory( src/concurrency MarlinCore_sources )
-aux_source_directory( plugins MarlinCore_sources )
-
 marlin_add_doxygen_sources( GLOB_RECURSE DIRECTORY include MATCHING_PATTERN *.h )
 marlin_add_doxygen_sources( GLOB_RECURSE DIRECTORY src MATCHING_PATTERN *.cc )
-marlin_add_doxygen_sources( GLOB_RECURSE DIRECTORY plugins MATCHING_PATTERN *.cc )
 
-add_library( MarlinCore SHARED ${MarlinCore_sources} )
-add_library( Marlin::Core ALIAS MarlinCore )
-install( TARGETS MarlinCore EXPORT Marlin LIBRARY )
+add_library( Core SHARED ${MarlinCore_sources} )
+add_library( Marlin::Core ALIAS Core )
+set_target_properties( Core PROPERTIES OUTPUT_NAME MarlinCore )
+install( TARGETS Core EXPORT Marlin LIBRARY )
 # Dependencies include dirs as system one to avoid unnecessary warnings
 target_include_directories( 
-  MarlinCore SYSTEM BEFORE 
-  PUBLIC ${streamlog_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIR}
+  Core SYSTEM PUBLIC 
+  $<BUILD_INTERFACE:${streamlog_INCLUDE_DIRS}> $<BUILD_INTERFACE:${TinyXML_INCLUDE_DIR}>
 )
 target_include_directories( 
-  MarlinCore BEFORE 
-  PUBLIC include
+  Core BEFORE PUBLIC 
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
 )
 target_link_libraries( 
-  MarlinCore PUBLIC
+  Core PUBLIC
   ${streamlog_LIBRARIES} 
   ${TinyXML_LIBRARY}
   ${CMAKE_DL_LIBS} 
   ${CMAKE_THREAD_LIBS_INIT}
 )
 set_target_properties( 
-  MarlinCore
+  Core
   PROPERTIES 
     COMPILE_FLAGS ${MARLIN_COMPILE_OPTIONS}
     VERSION ${${PROJECT_NAME}_VERSION}
     CXX_STANDARD ${MARLIN_CXX_STANDARD}
 )
 install( DIRECTORY include/marlin TYPE INCLUDE )
 
+
+# MarlinCorePlugins module
+aux_source_directory( plugins MarlinCorePlugins_sources )
+marlin_add_doxygen_sources( GLOB_RECURSE DIRECTORY plugins MATCHING_PATTERN *.cc )
+
+add_library( CorePlugins MODULE ${MarlinCorePlugins_sources} )
+add_library( Marlin::CorePlugins ALIAS CorePlugins )
+set_target_properties( CorePlugins PROPERTIES OUTPUT_NAME MarlinCorePlugins )
+install( TARGETS CorePlugins LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+
+set_target_properties( 
+  CorePlugins
+  PROPERTIES 
+    COMPILE_FLAGS ${MARLIN_COMPILE_OPTIONS}
+    VERSION ${${PROJECT_NAME}_VERSION}
+    CXX_STANDARD ${MARLIN_CXX_STANDARD}
+)
+target_link_libraries( 
+  CorePlugins PUBLIC Marlin::Core
+)
+
+# Add MarlinCorePlugins library to list of MARLIN_DLL libraries
+marlin_add_plugin_library( TARGETS CorePlugins )
+
 # Create all executables out of the main directory
 set( MarlinMT_BINARIES MarlinShellCompletion MarlinDumpPlugins Marlin MarlinMT )
 foreach( binary ${MarlinMT_BINARIES} )
   add_executable( bin_${binary} main/${binary}.cc )
   target_include_directories( 
     bin_${binary} SYSTEM BEFORE PUBLIC
-    $<TARGET_PROPERTY:MarlinCore,INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:Marlin::Core,INCLUDE_DIRECTORIES>
```

Comment:
You shouldn't have to call `target_include_directories`, because the non-private include_directories property of `Marlin::Core` should just be forwarded transitively when it is used in `target_link_library` a few lines below.
(Same for compile definitions, linked libraries etc.) 
Or are you running into issues of include location order?

Reply:
> Or are you running into issues of include location order?

No, I'm just learning new CMake :-)

> You shouldn't have to call target_include_directories, because the non-private > include_directories property of Marlin::Core should just be forwarded transitively when it is used in target_link_library a few lines below.
> (Same for compile definitions, linked libraries etc.)

Nice to know, I'll update it. But I'm running into huge troubles with conflicts. I'll first resolve it and implement these new changes.
Thanks.

Reply:
Ok. It is this transitivity of properties that makes using targets instead of variables so much more concise. But, yeah, it takes some time to get used to this approach, and the inconsistent implementation in other packages (root, geant4...) doesn't really help...